### PR TITLE
Ruby Group Capstone - Catalog of things [ TM#1 ]👨‍💻  

### LINKED ISSUE #1: [1pt] Add Options - [TM#1]
- The following options should be available:
     - List all books
     - List all labels (e.g. 'Gift', 'New')
     - Add a book

### LINKED ISSUE #3: [2pt] Create methods - [TM#1]
- Implement methods:
     - add_item method in the Label class
         - should take an instance of the Item class as an input
         - should add the input item to the collection of items
         - should add self as a property of the item object (by using the correct setter from the item object)
    - can_be_archived?() in the Book class
          - should override the method from the parent class
         - should return true if parent's method returns true OR if cover_state equals to "bad"
     - otherwise, it should return false

### LINKED ISSUE #7: [2pt] Create a Book Class Task1 [TM#1]
- Create a Book class in a separate .rb file.
- All Book class properties visible in the diagram should be defined and set up in the constructor method.

### LINKED ISSUE #10: [4pt] Implement startup actions - Task4 [TM#1]
- The following options should be available
-List of games
     - List all authors (e.g. 'Stephen King')
     - Add a game

### LINKED ISSUE #11: [1pt] Preserve Data - Task5 - [TM#1]
All data should be preserved by saving collections in .json files.

### LINKED ISSUE #15: [2pt] Add unit tests for all implemented methods - Task3 - [TM1]
- Add unit tests for all implemented methods.

### LINKED ISSUE #16: [4pt] Implement startup actions - Task4 - [TM#1]
- The following options should be available:
     - List all music albums
     - List all genres (e.g 'Comedy', 'Thriller')
     - Add a music album

### LINKED ISSUE #27: [1pt] Create a Label Class - [TM#1]
- Create a Label class with an association to the Item class (in a separate .rb file).
- All Label class properties visible in the diagram should be defined and set up in the constructor method.

### LINKED ISSUE #29: [1pt] Preserve Data - [TM#1]
- All data should be preserved by saving collections in .json files.

### LINKED ISSUE #32: [2pt] Implement methods - [TM#1]
- Implement methods:
     - add_item method in the Label class
         - should take an instance of the Item class as an input
         - should add the input item to the collection of items
         - should add self as a property of the item object (by using the correct setter from the item object)
     - can_be_archived?() in the Book class
         - should override the method from the parent class
         - should return true if parent's method returns true OR if cover_state equals to "bad"
         - otherwise, it should return false

### PR DESCRIPTION
**Pull request by tsheporamantso**
## On this milestone 🗿, I've managed to implement the following 🧠  
- Create a Book class in a separate .rb file.
- All Book class properties visible in the diagram should be defined and set up in the constructor method.
- Create a Label class with an association to the Item class (in a separate .rb file).
- All Label class properties visible in the diagram should be defined and set up in the constructor method.
-  Implement methods:
     - add_item method in the Label class
         - should take an instance of the Item class as an input
         - should add the input item to the collection of items
         - should add self as a property of the item object (by using the correct setter from the item object)
    - can_be_archived?() in the Book class
          - should override the method from the parent class
         - should return true if parent's method returns true OR if cover_state equals to "bad"
     - otherwise, it should return false
 - Implement methods:
     - add_item method in the Label class
         - should take an instance of the Item class as an input
         - should add the input item to the collection of items
         - should add self as a property of the item object (by using the correct setter from the item object)
     - can_be_archived?() in the Book class
         - should override the method from the parent class
         - should return true if parent's method returns true OR if cover_state equals to "bad"
         - otherwise, it should return false
  - The following options should be available:
     - List all music albums
     - List all genres (e.g 'Comedy', 'Thriller')
     - Add a music album
  - The following options should be available
  - List of games
     - List all authors (e.g. 'Stephen King')
     - Add a game
  - The following options should be available:
     - List all books
     - List all labels (e.g. 'Gift', 'New')
     - Add a book
  - All data should be preserved by saving collections in .json files.
  - Add unit tests for all implemented methods.
  - Create a schema.sql file with tables that will be analogical to the structure of the classes that you created:
     - books table (add all properties and associations from the parent Item class as table columns)
     - labels table

### CODE REVIEW

File: classes/book_details.rb [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    @books << {
+        'title' => title,
+        'publisher' => publisher,
+        'cover_state' => cover_state,
+        'publish_date' => publish_date
+    }
+
+    @labels << {
+        'title' => title
+        'color' => color
```
Comment (Rifat8080) [inline classes/book_details.rb]: Coma's missing after attributes values

**Review by Rifat8080 — COMMENTED** [review]
Hi @tsheporamantso 

Highlights
✔️ File structures are build well

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

Required Changes ♻️
Check the comments under the review.

Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

File: classes/book_details.rb (hunk 1 above)
Comment (tsheporamantso) [reply classes/book_details.rb]: Thank you for your insightful review, will make changes and resubmit.

**Review by Rifat8080 — APPROVED** [review]

**Comment by tsheporamantso** [comment]
#7, #24, #15,#10, #1, #10, #11, #2 , #3, #32, #33, #16

**Pull request merged**